app/scripts/lib/metrics.js

```diff
@@ -153,6 +159,66 @@ define(function (require, exports, module) {
       this._clearInactivityFlushTimeout();
     },
 
+    notifications: {
+      /* eslint-disable sorting/sort-object-props */
+      'flow.initialize': '_initializeFlowModel',
+      'flow.event': '_logFlowEvent'
+      /* eslint-enable sorting/sort-object-props */
+    },
+
+    /**
+     * Initialize the flow model. If it's already been initalized, do nothing.
+     * Initialization may fail if the required flow properties can't be found,
+     * either in the DOM or the resume token.
+     */
+    _initializeFlowModel () {
+      if (this._flowModel) {
+        return;
+      }
+
+      const flowModel = new Flow({
+        sentryMetrics: this._sentryMetrics,
+        window: this._window
+      });
+
+      if (flowModel.has('flowId')) {
+        this._flowModel = flowModel;
+      }
+    },
+
+    /**
+     * Log a flow event. If there is no flow model, do nothing.
+     *
+     * @param {Object} data
+     * Event data.
+     *
+     *   @param {String} data.event
+     *   The name of the event.
+     *
+     *   @param {String} [data.view]
+     *   The name of the view, to be interpolated to the event name.
+     *   If unset, the event will be logged without a view name.
+     *
+     *   @param {Boolean} [data.once]
+     *   If set, emit this event via the `logEventOnce` method.
+     *   Defaults to `false`.
+     */
+    _logFlowEvent (data) {
+      if (! this._flowModel) {
+        // If there is no flow model, we're not in a recognised flow
```

Comment:
This condition dates from pre-#4676 actually, so the comment should probably be slightly different now. I think the only case now is if the flow model hasn't initialised for some reason (so the DOM attributes are missing). I'll test to make sure my understanding is correct, then update the comment.